### PR TITLE
Fix box-box intersection algorithm with test

### PR DESCRIPTION
**Pull request by jslee02**
I recreated pull request #42 adding some tests. It wasn't able to add the tests into the original pull request since I don't have access to commit the author's forked repository.

It is confirmed that the tests [fail](https://travis-ci.org/dartsim/fcl/builds/58801002) without the fix.

### CODE REVIEW

File: src/narrowphase/narrowphase.cpp [hunk 1]
```diff
@@ -1492,14 +1503,19 @@ bool boxBoxIntersect(const Box& s1, const Transform3f& tf1,
...
+    // for(size_t i = 0; i < contacts.size(); ++i)
+    // {
+    //   contact_point += contacts[i].point;
+    // }
 
-    contact_point = contact_point / (FCL_REAL)contacts.size();
+    // contact_point = contact_point / (FCL_REAL)contacts.size();
 
-    *contact_points = contact_point;
+    // *contact_points = contact_point;
```
Comment (scpeters) [inline src/narrowphase/narrowphase.cpp]: Wouldn't it be better to delete this old code than leave it mysteriously commented?

**Comment by scpeters** [comment]
Can you resolve the conflicts?

File: src/narrowphase/narrowphase.cpp (hunk 1 above)
Comment (jslee02) [inline src/narrowphase/narrowphase.cpp]: Sure, removed by 366a915. I guess the original author (@EdsterG) would think so as well.

**Comment by jslee02** [comment]
Resolved. 

As a side note, the box-box collision algorithm needs to be changed to return multiple contact points instead of averaging them once #52 is merged.

**Comment by scpeters** [comment]
:+1:

**Pull request merged**